src/transport_picoquic.cpp

```diff
@@ -269,16 +270,20 @@ pq_loop_cb(picoquic_quic_t* quic, picoquic_packet_loop_cb_enum cb_mode, void* ca
 
                 packet_loop_time_check_arg_t* targ = static_cast<packet_loop_time_check_arg_t*>(callback_arg);
 
-                /*
-                log_msg << "packet_loop_time_check time: "
-                    << targ->current_time << " delta: " << targ->delta_t;
-                transport->logger.log(LogLevel::info, log_msg.str());
-                */
-
-                // TODO: Add config to set this value. This will change the loop select
-                //   wait time to delta value in microseconds. Default is <= 10 seconds
-                if (targ->delta_t > 1000) {
-                    targ->delta_t = 1000;
+                if(!transport->pq_datagram_ready) {
+                    transport->update_datagram_ready_status(picoquic_get_first_cnx(quic), false);
+                    targ->delta_t = 10000;
+                } else {
+                    if(targ->delta_t > 10000) {
+                        // ensure we send the queued data immediately
+                        if (transport->checkTxData()) {
+                            targ->delta_t = 0;
+                            transport->update_datagram_ready_status(picoquic_get_first_cnx(quic), true);
+                        } else {
+                            targ->delta_t = 10000;
+                            transport->update_datagram_ready_status(picoquic_get_first_cnx(quic), false);
```

Comment:
```suggestion
const bool enoughDelta = targ->delta_t > 10000;
targ->delta_t = enoughDelta ? 0 : 10000;
transport->update_datagram_ready_status(picoquic_get_first_cnx(quic), enoughDelta && transport->checkTxData());    
```
Obviously `enoughDelta` is a bad name, but I also have no idea what `targ->delta_t > 10000` means, so....

Reply:
Or what `delta_t` is, but that looks like it came from somewhere else.... the _t is confusing because in C/C++ that typically means "_type", but I am guessing they are going for "_time"?